packages/credential-eip712/src/agent/CredentialEIP712.ts

```diff
@@ -0,0 +1,321 @@
+      if(!credential.proof || !credential.proof.proofValue) throw new Error("Proof is undefined")
+      if(
+        !credential.proof.eip712 || 
+        !credential.proof.eip712.messageSchema ||
+        !credential.proof.eip712.domain 
+      ) throw new Error("eip712 is undefined");
+      
+      const { proof, ...signingInput } = credential;
+      const { proofValue, eip712, ...verifyInputProof} = proof;
+      const verificationMessage = {
+        ...signingInput,
+        proof: verifyInputProof
+      }
+  
+      const objectToVerify = {
+        message: verificationMessage,
+        domain: eip712.domain,
+        types: eip712.messageSchema,
+        primaryType: eip712.primaryType
+      }
+  
+      const recovered = recoverTypedSignature({
+        data: objectToVerify,
+        signature: proofValue,
+        version: SignTypedDataVersion.V4
+      })
+      
+      const issuer = extractIssuer(credential)
+      if (!issuer || typeof issuer === 'undefined') {
+        throw new Error('invalid_argument: args.credential.issuer must not be empty')
+      }
+  
+      const didDocument = await resolveDidOrThrow(issuer, context);
+
+      if (didDocument.verificationMethod) {
+        for(const verificationMethod of didDocument.verificationMethod) {
+          if (getEthereumAddress(verificationMethod)?.toLowerCase() === recovered.toLowerCase()) {
+            return true
+          }
+        }
+      }
+
+      throw new Error("Recovered Address does not match issuer")
+    } catch (e: any) {
+      throw new Error(e);
+    }
+  }
+
+    /** {@inheritdoc ICredentialIssuerEIP712.createVerifiablePresentationEIP712} */
+    async createVerifiablePresentationEIP712(
+      args: ICreateVerifiablePresentationEIP712Args,
+      context: IRequiredContext,
+    ): Promise<VerifiablePresentation> {
+      const presentationContext = processEntryToArray(
+        args?.presentation?.['@context'],
+        MANDATORY_CREDENTIAL_CONTEXT,
+      )
+      const presentationType = processEntryToArray(args?.presentation?.type, 'VerifiablePresentation')
+      let issuanceDate = args?.presentation?.issuanceDate || new Date().toISOString()
+      if (issuanceDate instanceof Date) {
+        issuanceDate = issuanceDate.toISOString()
+      }
+  
+      const presentation: PresentationPayload = {
+        ...args?.presentation,
+        '@context': presentationContext,
+        type: presentationType,
+        issuanceDate,
+      }
+  
+      if (!isDefined(args.presentation.holder)) {
+        throw new Error('invalid_argument: args.presentation.holder must not be empty')
+      }
+  
+      if (args.presentation.verifiableCredential) {
+        const credentials = args.presentation.verifiableCredential.map((cred) => {
+          // map JWT credentials to their canonical form
+          if (typeof cred !== 'string' && cred.proof.jwt) {
+            return cred.proof.jwt
+          } else {
+            return cred
```

Comment:
This likely needs to be serialized to a string to ensure that the credentials array uses a single data type:
```suggestion
            return JSON.stringify(cred)
```